### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**
[DEMO LINK](https://andriy-shymkiv.github.io/react_list-of-goods/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+  };
+
+  reverseButton = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  resetButton = () => {
+    this.setState({ sortType: SortType.NONE, isReversed: false });
```
Comment (Anna-Shapovalova) [inline src/App.tsx:87]:
```suggestion
    this.setState({ 
      sortType: SortType.NONE,
      isReversed: false,
    });
```
Suppose it`s better to format it like this to make the code clear and more readable

File: src/App.tsx [hunk 2]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sortType === SortType.ALPABET
+              ? 'button is-info'
+              : 'button is-info is-light'}
+            onClick={() => this.sortByAlphabetButton()}
```
Comment (Anna-Shapovalova) [inline src/App.tsx:105]:
```suggestion
            onClick={this.sortByAlphabetButton}
```

File: src/App.tsx [hunk 3]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={sortType === SortType.ALPABET
+              ? 'button is-info'
+              : 'button is-info is-light'}
```
Comment (Anna-Shapovalova) [inline src/App.tsx:104]: use here a classNames library

File: src/App.tsx [hunk 4]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+            onClick={() => this.sortByAlphabetButton()}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={sortType === SortType.LENGTH
+              ? 'button is-success'
+              : 'button is-success is-light'}
```
Comment (Anna-Shapovalova) [inline src/App.tsx:114]: a classNames library )

File: src/App.tsx [hunk 5]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+            onClick={() => this.sortByLengthButton()}
```
Comment (Anna-Shapovalova) [inline src/App.tsx:115]:
```suggestion
            onClick={this.sortByLengthButton}
```

File: src/App.tsx [hunk 6]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={isReversed
+              ? 'button is-warning'
+              : 'button is-warning is-light'}
```
Comment (Anna-Shapovalova) [inline src/App.tsx:124]: as well)

File: src/App.tsx [hunk 7]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+            onClick={() => this.reverseButton()}
```
Comment (Anna-Shapovalova) [inline src/App.tsx:125]:
```suggestion
            onClick={this.reverseButton}
```

File: src/App.tsx [hunk 8]
```diff
@@ -26,70 +26,125 @@ type ReorderOptions = {
...
+          >
+            Reverse
+          </button>
+
+          {(isReversed || sortType !== SortType.NONE) && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => this.resetButton()}
```
Comment (Anna-Shapovalova) [inline src/App.tsx:134]:
```suggestion
              onClick={this.resetButton}
```

**Review by Anna-Shapovalova — CHANGES_REQUESTED** [review]
Nice code))) 👍  Let`s fix some moments)

File: src/App.tsx (hunk 3 above)
Comment (Andriy-Shymkiv) [reply src/App.tsx:107]: classNames library looks giant in this case, but with ternar it looks pretty tiny, so i will keep ternar if you don't mind 😺

**Review by Anna-Shapovalova — APPROVED** [review]

File: src/App.tsx (hunk 3 above)
Comment (denys-danyliuk) [reply src/App.tsx:107]: it's better to use classNames)

**Pull request closed without merging**